roles/openshift-applier/tasks/process-template.yml

```diff
@@ -0,0 +1,64 @@
+---
+
+- name: "Fail if template but no params is specified"
+  fail:
+    msg: "Template specified, but no params file or variables supplied"
+  when:
+  - template|trim != ''
+  - params|trim == ''
+  - params_from_vars|trim == ''
+
+- name: "Check if params file exists (if applicable)"
+  stat:
+    path: "{{ tmp_inv_dir }}{{ params }}"
+  ignore_errors: yes
+  register: params_result
+  when:
+  - params|trim != ''
+
+- name: "Fail if params file doesn't exist (if applicable)"
+  fail:
+    msg: "{{ params }} - params file doesn't exist."
+  when:
+  - params|trim != ''
+  - params_result.stat.exists == False
+
+- name: "Determine location for the 'template' (if applicable)"
+  vars:
+    path: "{{ template }}"
+  import_tasks: check-file-location.yml
+  when:
+  - template|trim != ''
+
+- name: "Set 'template' values (if applicable)"
+  set_fact:
+    template: "{{ file_path }}"
+    template_f_option: "{{ option_f }}"
+  when:
+  - template|trim != ''
+
+- name: "Include any pre-processing role(s) before applying file and/or template"
+  include_tasks: pre-post-step.yml
+  with_items:
+  - "{{ content.pre_steps | default('') }}"
+  loop_control:
+    loop_var: step
+
+- name: "Process template with params from vars"
```

Comment:
Let's look for a solution to avoid having 2 different `include_tasks` for a "minor" difference such as params v.s. params_from_vars. IMHO they should be able to co-exist.

Reply:
Also, just as the `file` process, please review the use of `when` statements as much of that logic is no longer needed due to the split into a separate file.

Reply:
I'm more concerned about the readability and clear understanding of what is going on in the code here. Before, `template` and `file` were both in the same `process-one-entry` file, and it wasn't very clear what was doing on. Those two files were definitely done with my OOP hat on. I'm going to leave it where they are.

Reply:
ok, I'll assist in looking for a better solution.